README.md

```diff
@@ -28,6 +28,17 @@ If you wish to proxy only specific paths, you can use a router middleware to acc
 
 ### Options
 
+#### agent
+
+Use a custom `http.Agent` for proxy requests.
+
+```js
+var agent = new http.Agent(options);
+app.use(proxy('www.google.com', {
```

Comment:
Do we need the ability to specify one for `http` and one for `https`?

Reply:
I'm pretty sure you need a separate agent for https based on node docs: https://nodejs.org/api/https.html

What isn't clear to me is what happens if you set `https` option to `true` but pass in an http agent instead of https.

Reply:
I think it won't work if you pass in an `Agent` that's different from the `http` setting. I don't feel like it's required to handle that validation in the library, but if you do we can add it.

Right now `https.Agent` descends from `http.Agent`, and we require `agent` to be an `http.Agent` in the types, so using either passes the type check.

Reply:
👍  seems fair enough to leave it to the person using the proxy to specify an https agent if they watn it to handle https requests.